core/src/main/java/org/bitcoinj/quorums/ChainLocksHandler.java

```diff
@@ -64,21 +66,36 @@ public class ChainLocksHandler extends AbstractManager implements RecoveredSigna
     StoredBlock lastNotifyChainLockBlock;
 
     HashMap<Sha256Hash, Long> seenChainLocks;
+
+    // keep track of block hashes and clsig
+    LinkedHashMap<Sha256Hash, ChainLockSignature> chainlockMap = new LinkedHashMap<Sha256Hash, ChainLockSignature>() {
+        @Override
+        protected boolean removeEldestEntry(Map.Entry entry) {
+            return size() > 5000;
+        }
+    };
+    LinkedHashMap<Sha256Hash, ChainLockSignature> coinbaseChainlockMap = new LinkedHashMap<Sha256Hash, ChainLockSignature>() {
+        @Override
+        protected boolean removeEldestEntry(Map.Entry entry) {
+            return size() > 5000;
+        }
+    };
```

Comment:
This will be an in-memory-only hash table that associates a block with the coinbase chainlock signature.